### PR TITLE
feat: remove visibility from url's and api requests

### PR DESCRIPTION
**Pull request by seve**
### Reviewers
**Functional:** @tihuan && (@danieljhegeman || @Bento007)

---

## Changes
- modify
  – Remove visibility where applicable 

## Notes:

Note that I retained the old collection map for revision ↔ published collection relation.  There are probably some performance and request number optimizations to be made. But would rather get a feature complete PR out quick and then address performance opportunities later, if ever.

rdev: https://dan-buggy-2-frontend.rdev.single-cell.czi.technology/

### CODE REVIEW

File: frontend/src/common/queries/collections.ts [hunk 1]
```diff
@@ -281,44 +277,45 @@ async function collectionUploadLinks({
...
-  id = "",
-  visibility = VISIBILITY_TYPE.PRIVATE
-) {
+  collection: Collection | TombstonedCollection | null | undefined
+): UseMutationResult<
+  void,
+  unknown,
+  { collectionID: string },
+  { previousCollections: CollectionResponsesMap }
+> {
```
Comment (seve) [inline frontend/src/common/queries/collections.ts:313]: @tihuan Is this appropriate typing here?

Comment (tihuan) [reply frontend/src/common/queries/collections.ts:313]: discussion: Wondering if the interface could be the same primitive types, so the arguments are more predictable, and we can just set default `""` directly there? 

https://www.johndcook.com/blog/2011/07/19/you-wanted-banana/

The return type looks good!

E.g.,

```ts
function useDeleteCollection(id = "", visibility = ""): UseMutationResult<...>  { ... } 
```

**Review by tihuan — COMMENTED** [review]
Code LGTM! Just one quick discussion on `useDeleteCollection()` arguments! Thank you 🙆‍♂️

File: frontend/src/common/queries/collections.ts (hunk 1 above)
Comment (seve) [reply frontend/src/common/queries/collections.ts:314]: agreed, was trying out something new, but I think it makes sense this way

Comment (seve) [reply frontend/src/common/queries/collections.ts:314]: Note though, this introduced _a lot_ of prop drilling:
https://github.com/chanzuckerberg/single-cell-data-portal/pull/2197/commits/450bcb143c2e324a87798a21ed22408b4ce270f1

Comment (tihuan) [reply frontend/src/common/queries/collections.ts:314]: Woah 7 levels lol The deepest I've seen so far ahaha

Maybe time to pass `id` and `visibility` as context?

File: frontend/src/components/Collections/components/Grid/components/Row/CollectionRow/index.tsx [hunk 2]
```diff
@@ -85,10 +84,7 @@ const CollectionRow: FC<Props> = (props) => {
   return (
     <StyledRow data-test-id="collection-row">
       <StyledCell>
-        <Link
-          href={`/collections/${id}${isPrivate ? "/private" : ""}`}
-          passHref
-        >
+        <Link href={`/collections/${id}]`} passHref>
```
Comment (Bento007) [inline frontend/src/components/Collections/components/Grid/components/Row/CollectionRow/index.tsx:87]:
```suggestion
        <Link href={`/collections/${id}`} passHref>
```

**Review by Bento007 — CHANGES_REQUESTED** [review]

File: frontend/src/common/queries/collections.ts [hunk 3]
```diff
@@ -82,9 +84,10 @@ async function fetchCollections(): Promise<CollectionResponsesMap> {
   const collectionsMap: CollectionResponsesMap = new Map();
 
   for (const collection of json.collections as CollectionResponse[]) {
-    const collectionsWithId = collectionsMap.get(collection.id) || new Map();
-    collectionsWithId.set(collection.visibility, collection);
-    collectionsMap.set(collection.id, collectionsWithId);
+    const publicID = collection.revision_of || collection.id;
```
Comment (seve) [inline frontend/src/common/queries/collections.ts:87]:
```suggestion
    const publicID = collection.visibilty === Visibility.PRIVATE?  collection.revision_of : collection.id;
```

**Comment by Bento007** [comment]
When I start a revision I can immediately publish without making any changes
Screen Shot 2022-04-04 at 10 13 29 AM

**Comment by Bento007** [comment]
When uploading a dataset, the upload spinner transition between processing and uploading. Is there is state where a dataset goes back to uploading from processing?

**Comment by Bento007** [comment]
A refresh is required to show the deleted tag when deleting a dataset from a revision.

**Comment by danieljhegeman** [comment]
> When I start a revision I can immediately publish without making any changes Screen Shot 2022-04-04 at 10 13 29 AM

@Bento007 this should no longer be a problem!

**Review by danieljhegeman — APPROVED** [review]

**Pull request merged**